session1/solutions/assignment4/Rakefile

```diff
@@ -1 +1,46 @@
-# assignment 4 solution
+require 'pathname'
+require 'rubygems/package'
+
+desc 'create archive file from a path'
+task :archive do
+
+  archive_ext = "tar.gz"  
+  path = ENV["PATH2TAR"] || "."
+  exclude = ENV["EXCLUDE"] + "," + archive_ext || archive_ext
+
+  exclude_list = exclude.split(',')
+
+  # check if path exist and set the archive file name
+  if Dir.exist?(path) == false then
+    puts "== archive == Error: Path doesn't exist"
+    exit
+  else
+    Dir.chdir("#{path}")
+    puts "== archive == Archive requested for path: #{Dir.pwd}"
+    tar_name = Pathname.new(Dir.pwd).basename.to_s.gsub(/[^a-z,A-Z]/, '_') + archive_ext
+    puts "== archive == Archive will be saved in file: #{tar_name}"
+  end
+
+  # prepare file list and exclude unwanted files 
+  files = FileList.new("#{Dir.pwd.to_s}/*")
+  exclude_list.each do |e|
+    files.exclude(/#{e}$/)
+  end 
+
+  # write the requested files to a tar archive 
```

Comment:
This is a lot of code for something relatively simple. I would check if there are ruby/rake libraries that may save you many lines of coding